contracts/MetadataViews.cdc

```diff
@@ -415,4 +415,96 @@ pub contract MetadataViews {
+    //
+    // This is used to get traits of individual key/value pairs along with some contextualized data about the trait
+    pub struct Trait {
+        // The name of the trait. Like Background, Eyes, Hair, etc.
+        pub let traitType: String
+
+        // The underlying value of the trait, the rest of the fields of a trait provide context to the value.
+        pub let value: AnyStruct
+
+        // displayType is used to show some context about what this traitType and value represent
+        // for instance, you could set value to a unix timestamp, and specify displayType as "Date" to tell
+        // platforms to consume this trait as a date and not a number
+        pub let displayType: String?
+
+        // Rarity can also be used directly on an attribute.
+        //
+        // This is optional because not all attributes need to contribute to the NFT's rarity.
+        pub let rarity: Rarity?
+
+        init(traitType: String, value: AnyStruct, displayType: String?, rarity: Rarity?) {
+            self.traitType = traitType
+            self.value = value
+            self.displayType = displayType
+            self.rarity = rarity
+        }
+    }
+
+    // A view to return all the traits on an NFT.
+    //
+    // This is used to get traits of individual key/value pairs along with some contextualized data about the trait
+    pub struct Traits {
+        pub let traits: [Trait]
+
+        init(_ traits: [Trait]) {
+            self.traits = traits
+        }
+
+        pub fun addTrait(_ t: Trait) {
+            self.traits.append(t)
+        }
+    }
+
+    // helper function to easily convert a dictionary to traits. For NFT collections that do not need either of the
+    // optional values of a Trait, this method should suffice to give them an array of valid traits
+    pub fun dictToTraits(dict: {String: AnyStruct}, excludedTraitTypes: [String]?): Traits {
+        // Collection owners might not want all the fields in their metadata included.
+        // They might want to handle some specially, or they might just not want them included at all.
+        let excludedTraits: {String: Bool} = {}
+        if excludedTraitTypes != nil {
+            for k in excludedTraitTypes! {
+                excludedTraits[k] = true
+            }
+        }
+
+        let traits: [Trait] = []
+        for k in dict.keys {
+            // this key is part of our excluded traits set, skip it
+            if excludedTraits[k] != nil {
+                continue
+            }
+
+            let trait = Trait(traitType: k, value: dict[k]!, displayType: nil, rarity: nil)
+            traits.append(trait)
+        }
+
+        return Traits(traits)
+    }
+
+    /// Rarity information for a single rarity
+    //
+    /// Note that a rarity needs to have either score or description but it can have both
+    pub struct Rarity {
+        /// The score of the rarity as a number
+        ///
+        pub let score: UFix64?
```

Comment:
Should the `Rarity` struct make any attempt to standardize a score's scale?

e.g. I could imagine something like `maxScore = 100` so that score is rendered as `42/100`

Reply:
I think it would be wise to add something like that. That greater should be more rare and the max should be 100.0